devito/passes/clusters/factorization.py

```diff
@@ -78,8 +78,11 @@ def collect_const(expr):
             # Back to the running example
             # -> (a + c)
             add = Add(*v)
-            # -> 3.*(a + c)
-            mul = Mul(k, add, evaluate=False)
+            if add == 0:
```

Comment:
Note for reviewers: these are small bug fixes... without, in some corner circumstances we were generating expressions such as `a + 0`, while now we would generate just `a`